### PR TITLE
create ConnectionState before adding to HostMap

### PR DESCRIPTION
**Pull request by wadey**
We have a few small race conditions with creating the HostInfo.ConnectionState
since we add the host info to the pendingHostMap before we set this
field. We can make everything a lot easier if we just add an "init"
function so that we can set this field in the hostinfo before we add it
to the hostmap.

We have seen this issue cause crashes in v1.4.0.

### CODE REVIEW

File: handshake_manager.go [hunk 1]
```diff
@@ -191,13 +191,13 @@ func (c *HandshakeManager) handleOutbound(vpnIp iputil.VpnIp, f udp.EncWriter, l
 	}
 }
 
-func (c *HandshakeManager) AddVpnIp(vpnIp iputil.VpnIp) *HostInfo {
-	hostinfo := c.pendingHostMap.AddVpnIp(vpnIp)
-	// We lock here and use an array to insert items to prevent locking the
```
Comment (nbrownus) [inline handshake_manager.go:196]: I think if we remove this then we should add the locking or rename the function to `UnlockedAddVpnIp`

File: handshake_manager_test.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func Test_NewHandshakeManagerTrigger(t *testing.T) {
 
 	assert.Equal(t, 0, testCountTimerWheelEntries(blah.OutboundHandshakeTimer))
 
-	hi := blah.AddVpnIp(ip)
+	hi := blah.AddVpnIp(ip, nil)
```
Comment (nbrownus) [inline handshake_manager_test.go:74]: Lets add a test case to ensure our init func is called and the created return bool is correct

File: handshake_manager.go (hunk 1 above)
Comment (wadey) [reply handshake_manager.go:196]: This comment is outdated and from before the lock was moved to inside of the OutboundHandshakeTimer itself. Now we can finally delete this old and wrong comment!

File: handshake_manager_test.go (hunk 2 above)
Comment (wadey) [reply handshake_manager_test.go:86]: Test added higher up in the file in latest commit

**Review by nbrownus — APPROVED** [review]
Do we want to note this one in the changelog?

**Comment by wadey** [comment]
Added CHANGELOG

**Review by nbrownus — APPROVED** [review]

**Pull request merged**